Refactor Summaries

ersonp opened the pull request:
Did you run `make format && make check`?
yes

Fixes #757 

 Changes:	
- Changed `Summary` to `Overview` and `ExtraSummary` to `Summary`.
- Factored out TCPAddr and Online fields into its own struct `NetworkStats`.
- Removed `extraSummaryWithDmsgResp` and updated the new `Summary` accordingly.
- Renamed `getVisorsExtraSummary` to `getAllVisorsSummary`.
- Removed ` TCPAddr` field from `makeSummaryResp`

How to test this PR:
1. Run `make build-ui; make move-built-frontend ; make build; make install`
2. Start Hypervisor and check ui.

ersonp committed:
wip

ersonp committed:
Refactor

ersonp committed:
Refactor code

ersonp commented:
`dmsg stats` is needed by both `getVisorSummary` and `getAllVisorsSummary`.

jdknives commented on pkg/visor/api.go:
```diff
@@ -78,8 +78,8 @@ type HealthCheckable interface {
 
-// Summary provides a summary of a Skywire Visor.
-type Summary struct {
+// Overview provides a overview of a Skywire Visor.
```
Maybe be bit more verbose `Overview provides a range of basic information about a Visor`

ersonp committed:
Remove tcpaddr

ersonp committed:
Refactor

jdknives reviewed (changes requested):
Seems to me like TCP address is not actually completely removed. If it is, then we can probably get rid of the whole network stats structure because

1. `TCPAddr` is gone
2. `Online` field is not actually filled by visor iirc so it does not need to be part of the summary.

ersonp commented:
`"/visors", getVisors()` and `"/visors/{pk}", getVisor()` uses `NetworkStat` but i do not know where the endpoints are used. I removed `NetworkStat` from them and everything still works but I will need more info on them. `Online` is used by `getAllVisorsSummary()` and the port in `TCPAddr` is used by `getVisorSummary` so I replaced `TCPAddr` with `Port`.  Both `getAllVisorsSummary()` and `getVisorSummary` user `Overview` so `Online` and `Port` can't be taken out.

ersonp committed:
Remove networkstats

ersonp committed:
Revert "Remove networkstats"

This reverts commit de3dd9e79144a7a3595021acce4c01d3315e33f7.

ersonp committed:
Remove networkstats

jdknives commented on pkg/visor/api.go:
```diff
@@ -114,38 +114,41 @@ func (v *Visor) Summary() (*Summary, error) {
+	Routes       []routingRuleResp              `json:"routes"`
+	IsHypervisor bool                           `json:"is_hypervisor,omitempty"`
+	DmsgStats    *dmsgtracker.DmsgClientSummary `json:"dmsg_stats"`
+	Port         uint16                         `json:"port"`
```
Not sure what we display that for. Is it the `dmsg` port?

ersonp replied:
Yes it's the `dmsg` port.

ersonp committed:
Remove dmsg port from summary

Senyoret1 reviewed (commented):
Working well, just 2 things:
- With the changes made, some parts of the UI code are no longer needed, but I checked and added them to a list, so I can make the changes after this PR is merged, to avoid delays with this one.
- I'm not a fan of updating the compiled UI files in PRs that were made for adding other changes, but if nobody else has problems with this, there is no problem.

ersonp committed:
Revert "Refactor"

This reverts commit 155d9fb

ersonp committed:
Revert "Remove networkstats"

This reverts commit de3dd9e79144a7a3595021acce4c01d3315e33f7.
